### PR TITLE
[feat] wikimedia commons: support for videos, audio and other files

### LINKED ISSUE #1221: Wikimedia Commons
**Working URL to the engine**
https://commons.wikimedia.org

**Why do you want to add this engine?**
Out of all of the Wikimedia projects, Wikimedia Commons is one of only two to not appear in any engine category in SearXNG, with the other being Wikispecies.

**Features of this engine**
It has a collection of [82,886,704](https://commons.wikimedia.org/wiki/Special:Statistics) [freely usable](https://commons.wikimedia.org/wiki/Commons:Reusing_content_outside_Wikimedia) media files.

**How can SearXNG fetch the information from this engine?**
`https://commons.wikimedia.org/w/index.php?search=%s` with `%s` being what you want to search.

**Applicable category of this engine**
General, files, images, music, videos.

### PR DESCRIPTION
**Pull request by Bnyro**
## What does this PR do?
- I've already implemented wikimedia commons support in https://github.com/searxng/searxng/pull/2588 - this PR finishes support for it

## How to test this PR locally?
- !wcv foo
- !wca foo
- !wcf foo

## Related issues
closes  #1221

### CODE REVIEW

File: searx/settings.yml [hunk 1]
```diff
@@ -1910,6 +1910,28 @@ engines:
...
+    engine: wikicommons
+    shortcut: wca
+    categories: music
+    search_type: audio
+    number_of_results: 10
+
+  - name: wikicommons.files
+    engine: wikicommons
+    shortcut: wcf
+    categories: files
```
Comment (return42) [inline searx/settings.yml:1933]: Not sure these are really "files" .. like we know from torrents by example.  TBH I don't know a better category for this content.

**Review by return42 — APPROVED** [review]
Thanks for completing the series of wikicommon engines :+1: 

note: all wikimedia-commons engines are enabled by default

@unixfox I remember you had some doubts about to many engines enabled by default (can't find the posts right now) ..

File: searx/settings.yml (hunk 1 above)
Comment (Bnyro) [reply searx/settings.yml:1933]: Yes, me neither. I've thought rather long about this too when implementing, the only choices I had were the `files` category or no category (`other`), so I decided that it makes sense to use the `files` category given the fact all results are files actually :)

**Comment by unixfox** [comment]
> @unixfox I remember you had some doubts about to many engines enabled by default (can't find the posts right now) ..

I'm a bit more conservative about the "general" category. I guess for the other ones it's ok because it's better to not bloat them too much.

**Pull request merged**